Makefile.am

```diff
@@ -151,12 +151,12 @@ clean-venv:
 	fi
 
 .PHONY: tests
-tests: tox
-	tox -e "tests,lib,network,virt,gluster,hooks"
+tests: tox venv
+	. $(VENV)/activate && tox -e "tests,lib,network,virt,gluster,hooks"
```

Comment:
Does it change the current shell?

I'm not sure it is a good idea to enter the venv temporarily for running the tests.

Having an easy way to create the venv is nice, simplifying life for new contributors.
But for developing, you really want to enter the venv and stay there for long time
(days?) so you edit and test quickly.

Also in the CI, we don't need the venv, since we install tox and the other
requirements in the container, so recreating the venv is just waste of time
and resources, and reason for failure (e.g. if pypi is not accessible).

I would not add a dependency or activation as part of running the tests.

Reply:
OK, for the CIs this is really not a good thing, true. I'll remove that dependency.